### PR TITLE
Add Namespace Dropdown to Gateway UI

### LINKED ISSUE #1379: [Feature] Namespace selector in UI
[Feature] Namespace selector in UI

## Expected Behaviour

As part of https://github.com/openfaas/faas-netes/issues/511 we can now deploy functions to multiple namespaces via the CLI, so it would be good to be able to show those in the main angular UI that OpenFaaS uses.

## Current Behaviour

Not available

## Possible Solution

A simple solution would be a good move - a new drop-down somewhere easy to find like in the left hand navigation / listing.

The /system/namespaces endpoint will return a list of namespaces to offer up.

### PR DESCRIPTION
**Pull request by Waterdrips**
## Description
This PR adds a namespace dropdown for the openfaas
namespaces. If a user has non-default namespaces
installed then a user can select that namespace,
view functions in that namespace, invoke them from
the UI and delete them from that namespace.

Signed-off-by: Alistair Hey <alistair@heyal.co.uk>

closes #1379 
## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
- [x] My issue has received approval from the maintainers or lead with the `design/approved` label
#1379 
Adding this allows a user to see, delete and invoke functions in another opoenfaas controlled namespace.

This PR does not add the ability to create a fn in that namespace, That can be added in a future PR
<br>
<br>
NS-Dropdown
Screenshot 2019-11-24 at 16 07 45
<br><br>
Different OF Namespace selected
Screenshot 2019-11-24 at 16 07 48

<br><br>
Function in another namespace has its Ns appended to the url
Screenshot 2019-11-24 at 16 07 55
<br><br>
Invoked function, showing it works 
Screenshot 2019-11-24 at 16 08 08

<br><br>
default namespace selected
Screenshot 2019-11-24 at 16 07 41

## How Has This Been Tested?
Deployed to my OpenFaaS instance and manually tested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Thank you for working on this.

I find the `Namespace(s)` being in each item looks a little odd. What if it was a caption and then put underneath the deployment button?

We also need `namespace` in the deploy pop-up dialog.

**Comment by Waterdrips** [comment]
New functionality allowing a user to deploy a function from the UI (store or custom) by using a new dropdown in the custom tab.

Also added the changes to the ns dropdown on the left bar that @alexellis requested.
See screenshots:

Screenshot 2019-11-25 at 08 10 48
Screenshot 2019-11-25 at 08 10 43
Screenshot 2019-11-25 at 08 10 34
Screenshot 2019-11-25 at 08 10 30

**Comment by alexellis** [comment]
Have you got a Docker image that I can swap into my local environment to test it out?

**Comment by Waterdrips** [comment]
`heyal/gateway:l34`

**Comment by alexellis** [comment]
If we only have one namespace, should it default?

Screenshot 2019-11-25 at 10 24 07

**Comment by alexellis** [comment]
Thanks

Should the drop-down be shown when there is only 1 namespace? This will be the case on Swarm, where we have no namespace support at all. Possibly on other providers.

File: gateway/assets/script/bootstrap.js [hunk 1]
```diff
@@ -52,15 +55,26 @@ app.controller("home", ['$scope', '$log', '$http', '$location', '$interval', '$f
...
         };
 
         $scope.invocation.request = "";
+
         var fetchFunctionsDelay = 3500;
         var queryFunctionDelay = 2500;
-        
+
+        var fetchNamespacesInterval = function() {
+            $http.get("../system/namespaces")
```
Comment (alexellis) [inline gateway/assets/script/bootstrap.js]: Hmm I didn't expect to see `../`?

**Pull request merged**